spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/PropertySourceUtils.java

```diff
@@ -54,20 +59,15 @@
 			return properties;
 		}
 		catch (IOException e) {
-			throw new IllegalArgumentException();
+			throw new UncheckedIOException(e);
 		}
 	};
 
 	/**
 	 * Function to convert Properties to a Map.
 	 */
 	public static final Function<Properties, Map<String, Object>> PROPERTIES_TO_MAP = p -> p.entrySet().stream()
-			.collect(Collectors.toMap(e -> String.valueOf(e.getKey()), Map.Entry::getValue, throwingMerger(),
-					java.util.LinkedHashMap::new));
-
-	private PropertySourceUtils() {
```

Comment:
not deleted, but moved at the top. much easier to read the code, especially for a utility class. at least, imo.